### ISSUE TITLE
识别模型NRTR能用于中文吗？

### ISSUE THREAD

**Issue by Lebron-Harden** [PaddlePaddle/PaddleOCR#4839]
如题，我尝试了用NRTR训练中文识别，训练数据是古籍文档，但是训练了二三十轮，acc还是很低，请问是NRTR本身就不适合训练中文呢，还是我设置的参数可能有问题？
![image](https://user-images.githubusercontent.com/74600923/144734129-b44a2eda-a047-462f-97c4-378f4308fdfc.png)

**Comment by Topdu** [comment]
在内部的中文数据集上nrtr是可以正常训练的，评估精度可以达到77%；
具体训练配置需要根据数据集的情况进行适配，我们的中文数据集字符数比较多，样本的长宽比例变得更大，因此会将输入调整为32*320，数据量少，要加上数据增强等等，您可以提供多一些数据集的信息，以便分析。

**Comment by Lebron-Harden** [comment]
> 在内部的中文数据集上nrtr是可以正常训练的，评估精度可以达到77%； 具体训练配置需要根据数据集的情况进行适配，我们的中文数据集字符数比较多，样本的长宽比例变得更大，因此会将输入调整为32*320，数据量少，要加上数据增强等等，您可以提供多一些数据集的信息，以便分析。

config文件如下，长宽比是我统计了数据集之后设置的，应该没问题

Global:
  use_gpu: True
  epoch_num: 300
  log_smooth_window: 20
  print_batch_step: 1
  save_model_dir: ./output/rec_nrtr/
  save_epoch_step: 100
  # evaluation is run every 2000 iterations after the 0th iteration
  eval_batch_step: [0, 1500]
  cal_metric_during_train: True
  pretrained_model: /home/dachuang/workspace/Pjz/PaddleOCR-2.3/pretrained/rec_nrtr/best_accuracy
  checkpoints: ./output/rec_nrtr/latest
  save_inference_dir:
  use_visualdl: False
  infer_img: 
  # for data or label process
  character_dict_path: ./ppocr/utils/MTH_dict.txt
  character_type: ch
  max_text_length: 34
  infer_mode: False
  use_space_char: True
  save_res_path: 

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.99
  clip_norm: 5.0
  lr:
    name: Cosine
    learning_rate: 0.001
    #warmup_epoch: 2
  regularizer:
    name: 'L2'
    factor: 0.00004

Architecture:
  model_type: rec
  algorithm: NRTR
  in_channels: 1
  Transform:
  Backbone:
    name: MTB
    cnn_num: 2
  Head:
    name: Transformer
    d_model: 512
    num_encoder_layers: 6
    beam_size: 0 # When Beam size is greater than 0, it means to use beam search when evaluation.
    dst_vocab_size: 6582
    

Loss:
  name: NRTRLoss
  smoothing: True

PostProcess:
  #name: CTCLabelDecode
  name: NRTRLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: SimpleDataSet
    data_dir: /home/dachuang/workspace/Pjz/Easiest-realization-CRNN-for-Chinese-master/MTH_v2/train
    label_file_list: ["./train_data/MTH_v2/8wtrain.txt"]
    transforms:
      - NRTRDecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - NRTRLabelEncode: # Class handling label
      #- CTCLabelEncode:
      - NRTRRecResizeImg:
          image_shape: [320, 32]
          resize_type: PIL # PIL or OpenCV
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: True
    batch_size_per_card: 128
    drop_last: False
    num_workers: 0

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: /home/dachuang/workspace/Pjz/Easiest-realization-CRNN-for-Chinese-master/MTH_v2/train
    label_file_list: ["./train_data/MTH_v2/2wtest.txt"]
    transforms:
      - NRTRDecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - NRTRLabelEncode: # Class handling label
      #- CTCLabelEncode:
      - NRTRRecResizeImg:
          image_shape: [320, 32]
          resize_type: PIL # PIL or OpenCV
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 256
    num_workers: 0
    use_shared_memory: False

**Comment by Lebron-Harden** [comment]
> 在内部的中文数据集上nrtr是可以正常训练的，评估精度可以达到77%； 具体训练配置需要根据数据集的情况进行适配，我们的中文数据集字符数比较多，样本的长宽比例变得更大，因此会将输入调整为32*320，数据量少，要加上数据增强等等，您可以提供多一些数据集的信息，以便分析。

训练的图片基本上是下面这种样式
![image](https://user-images.githubusercontent.com/74600923/144966530-24cd0afb-387c-4cd9-9c48-8b12d8f9c9fc.png)

**Comment by Topdu** [comment]
https://github.com/PaddlePaddle/PaddleOCR/blob/release%2F2.3/ppocr/modeling/heads/rec_nrtr_head.py#L168
在这个代码中默认解码字符的长度为25，应该修改为配置文件中的max_text_length: 34，其次，多尝试几次不同的学习率和正则参数

**Comment by Lebron-Harden** [comment]
> 

这个解码长度为34应该是可行的，我用其他模型跑的时候都是设置的34，没啥问题。另外，学习率我也尝试过不同的，也没太大用。主要是这个准确率在训练的时候都一直没有超过5%，请问还有可能是哪里出了问题？

**Comment by Lebron-Harden** [comment]
> https://github.com/PaddlePaddle/PaddleOCR/blob/release%2F2.3/ppocr/modeling/heads/rec_nrtr_head.py#L168 在这个代码中默认解码字符的长度为25，应该修改为配置文件中的max_text_length: 34，其次，多尝试几次不同的学习率和正则参数

这个解码长度为34应该是可行的，我用其他模型跑的时候都是设置的34，没啥问题。另外，学习率我也尝试过不同的，也没太大用。主要是这个准确率在训练的时候都一直没有超过5%，请问还有可能是哪里出了问题？

**Comment by Topdu** [comment]
https://github.com/PaddlePaddle/PaddleOCR/blob/release%2F2.3/ppocr/modeling/heads/rec_nrtr_head.py#L168的内容如下：
for len_dec_seq in range(1, 25):
是直接写死了25个字符，请改成34试试，训练的时候，打印一下预测结果和label的对比，看看是否有什么问题。

**Comment by Lebron-Harden** [comment]
> [https://github.com/PaddlePaddle/PaddleOCR/blob/release%2F2.3/ppocr/modeling/heads/rec_nrtr_head.py#L168的内容如下：](https://github.com/PaddlePaddle/PaddleOCR/blob/release%2F2.3/ppocr/modeling/heads/rec_nrtr_head.py#L168%E7%9A%84%E5%86%85%E5%AE%B9%E5%A6%82%E4%B8%8B%EF%BC%9A) for len_dec_seq in range(1, 25): 是直接写死了25个字符，请改成34试试，训练的时候，打印一下预测结果和label的对比，看看是否有什么问题。

奥明白，我试一下。感谢回复
